### ISSUE TITLE
pylint should warn about `type(x) == Y` and `type(x) is Y`

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#299]
Originally reported by: **Chris Rebert (BitBucket: [cvrebert](http://bitbucket.org/cvrebert), GitHub: @cvrebert?)**

---

People who are new to Python often forget that the `isinstance()` built-in function exists, and instead end up writing their typechecks non-idiomatically, using `type()` and an equality comparison. Here's an example:

```
#!python

def dispatch(x):
    if type(x) == int:
        print "int"
    elif type(x) is dict:
        print "dict"
```

pylint doesn't currently emit any relevant warnings for this code. I propose it should emit warnings suggesting that the code be rewritten to use `isinstance()` instead.

Corrected example:

```
#!python

def typecheck(x):
    if isinstance(x, int):
        print "int"
    elif isinstance(x, dict):
        print "dict"
```

(Yes, I know that these aren't _strictly_ equivalent due to subclasses and metaclasses, but those cases are pretty uncommon.)

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/299
